workers/unity/Packages/com.improbable.gdk.core/Systems/CleanReactiveComponentsSystem.cs

```diff
@@ -71,27 +60,32 @@ private void GenerateComponentGroups()
                     }
 
                     typesToRemove.Add(type);
-                    addRemoveComponentActionMethod.MakeGenericMethod(type).Invoke(this, null);
+                    componentGroupsToRemove.Add((GetComponentGroup(ComponentType.ReadOnly(type)), type));
                 }
             }
         }
 
-        private void AddRemoveComponentAction<T>()
+        private void RemoveComponents()
```

Comment:
Mild preference - can we move this after update. Was previously before because it was a use once function used only in CreateManager. Now it's used in update it would be nice if it was below that.
Our usual convention is actually private after protected though. Could also make these local functions. https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions